src/lib/authentication/components/protectedRoute.tsx

```diff
@@ -0,0 +1,36 @@
+import { useNavigate, Route } from "react-router-dom";
+import { useAuthContext } from "./useAuthContext";
+
+type ProtectedRouteProps = {
+  path: string;
+  element: React.ReactElement;
+  permissions: string[];
+  fallbackPath?: string;
+};
+
+const ProtectedRoute = ({
```

Comment:
I don't understand the purpose of this. 

This change the <Routes> of the application routes which is not the purpose.
Like for <AuthOnly>, some Routes may be wrapped into HasPermission and fallback should be <Unauthorized> by default.

This component react only on changes to the 'context' and not to the User profile and thus the Permissions so I'm not sure how it works.